rfcs/text/0428-extending-and-duplicating-types.md

```diff
@@ -206,19 +206,27 @@ Furthermore, resolving the intentions of the existing definition for the message
 
 We propose that we should treat type updates as complete replacements, so implementation is much more straightforward for embedding applications. Partial schema updates also add some level of indirection, and that may obfuscate error sources and error reasons. Therefore, updates to types must be complete replacements rather than partial schema updates.
 
+### Changes to the link type system
+
+The discussed extension to the type system has implications on link types which extend from each other. Given the above `Person` and saying it defines `Knows` and `Has Friend` links, and that those links are **ordered**. Currently, the order of a link is defined in a given direction, e.g. _left-to-right_, and has a single value. With the inheritance of link types this would presently imply, that the order of the `Knows` link _within the `Knows` links list_ would have to be the same as the order of the `Has Friend` link _within the `Has Friend` links list_.
+
+**To mitigate this problem, we decided to remove link ordering**. This means, that links will no longer have an inherent order and no assumption is being made of the order.
+
+This change will affect any code that relies on the order property of link types. If applications have such code, they will need to update it to use a different mechanism for ordering links. A possibility is to add an ordering property to the link type itself. This makes the type-system itself easier to reason about whereas it will also allow application to use other types for their ordering instead of being forced to use integral values.
+
 # Reference-level explanation
 
 [reference-level-explanation]: #reference-level-explanation
 
 ## Defining extended entity types
 
-In the Block Protocol, we will allow type extension through the `allOf` JSON Schema keyword which applies all constraints from the sub-schemas within the `allOf` array..
+In the Block Protocol, we will allow type extension through the `allOf` JSON Schema keyword which applies all constraints from the sub-schemas within the `allOf` array.
 
 We'll add the following fields to the existing Entity Type meta schema definition:
 
 ```json
 {
-  "$id": "https://blockprotocol.org/types/modules/graph/0.3.1/schema/entity-type",
+  "$id": "https://blockprotocol.org/types/modules/graph/0.3/schema/entity-type",
```

Comment:
The patch version was on purpose. The RFC would result in a new version of the spec. Although it was previously backwards compatible, but now with the removal of link ordering it will not be, so we might want to change this to `0.4`